tests/test_sequences.py

```diff
@@ -651,3 +651,57 @@ def child():
         assert cap.re_compiled.match(t.cub1) is None
 
     child()
+
+
+def test_truncate(all_terms):
+    """Test terminal.truncate and make sure it agrees with terminal.length"""
+    @as_subprocess
+    def child(kind):
+        from blessed import Terminal
+        term = Terminal(kind)
+
+        test_string = term.red("Testing ") + term.yellow("makes ") +\
+            term.green("me ") + term.blue("feel ") +\
+            term.indigo("good") + term.normal
+        stripped_string = term.strip_seqs(test_string)
+        for i in range(len(stripped_string)):
+            test_l = term.length(term.truncate(test_string, i))
+            assert test_l == len(stripped_string[:i])
+        test_nogood = term.red("Testing ") + term.yellow("makes ") +\
+            term.green("me ") + term.blue("feel ") +\
+            term.indigo("") + term.normal
+        trunc = term.truncate(test_string, term.length(test_string) - len("good"))
+        assert trunc == test_nogood
+
+    child(all_terms)
+
+
+def test_truncate_wide_end(all_terms):
+    """Ensure that terminal.truncate has the correct behaviour for wide characters."""
+    @as_subprocess
+    def child(kind):
+        from blessed import Terminal
+        term = Terminal(kind)
+        test_string = u"AB\uff23"  # ABＣ
+        assert term.truncate(test_string, 3) == u"AB"
+
+    child(all_terms)
+
+
+def test_truncate_wcwidth_clipping(all_terms):
+    """Ensure that terminal.truncate has the correct behaviour for wide characters."""
+    @as_subprocess
+    def child(kind):
+        from blessed import Terminal
+        term = Terminal(kind)
+        test_emptystring = ""
+        test_string = term.blue(u"one\x01two")
+        test_bsstring = term.bold(u"one\btwo")
+        assert term.truncate(test_emptystring, 4) == ""
+        assert term.truncate(test_string, 4) == term.blue(u"one\x01t")
+        # this is incorrect, i suppose, in that \b doesn't resolve as
+        # an erasing width of -1, I think the wcwidth library should
+        # handle that, though!
```

Comment:
actually, `\b` interpretation does belong in blessed, and it is, by the `padd` function,
in an example sequence that contains `Terminal().move_right` sequence we would also truncate wrong
```print(t.truncate(t.bright_blue('yak ') + t.move_right(3) + t.bright_red('shaving'), 5))```
displays 8 columns, instead of 5, will add this change & test